cloudinit/distros/netbsd.py

```diff
@@ -0,0 +1,239 @@
+# Copyright (C) 2014 Harm Weites
+# Copyright (C) 2019 Gonéri Le Bouder
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import crypt
+import os
+import platform
+import six
+
+from cloudinit import distros
+from cloudinit import helpers
+from cloudinit import log as logging
+from cloudinit import net
+from cloudinit import ssh_util
+from cloudinit import util
+from cloudinit.distros import bsd_util
+
+LOG = logging.getLogger(__name__)
+
+
+class Distro(distros.Distro):
+    hostname_conf_fn = '/etc/rc.conf'
+    ci_sudoers_fn = '/usr/pkg/etc/sudoers.d/90-cloud-init-users'
+
+    def __init__(self, name, cfg, paths):
+        distros.Distro.__init__(self, name, cfg, paths)
+        # This will be used to restrict certain
+        # calls from repeatly happening (when they
+        # should only happen say once per instance...)
+        self._runner = helpers.Runners(paths)
+        self.osfamily = 'netbsd'
+        cfg['ssh_svcname'] = 'sshd'
+
+    def _select_hostname(self, hostname, fqdn):
+        if fqdn:
+            return fqdn
+        return hostname
+
+    def _read_system_hostname(self):
+        sys_hostname = self._read_hostname(filename='/etc/rc.conf')
+        return ('/etc/rc.conf', sys_hostname)
+
+    def _read_hostname(self, filename, default=None):
+        return bsd_util.get_rc_config_value('hostname')
+
+    def _write_hostname(self, hostname, filename):
+        bsd_util.set_rc_config_value('hostname', hostname,
+                                     fn='/etc/rc.conf')
+
+    def create_group(self, name, members):
```

Comment:
this whole function reads a bit confusing

maybe change "name" to "group"…?

Reply:
ok

Reply:
+1 on more context on variable/param names in general, but I realize that issue exists in Distro parent too :/ So no need to fix this in this branch as we should align the call signatures for every Distro subclass.

Note though that your method signature is different from top-level Distro.create_group which allows for members=None default. This method should match thiat

Reply:
Also this function is a near copy of Distro.create_group with two exceptions:
 1.  a util.system_is_snappy() check 
 2.  util.subp(['usermod', '-G', name, member])   here instead  of  '[usermod' '-a' '-G' name, member]

Though I think the is_snappy check shouldn't cause problems for you, we might be able to just define a user_mod_cmd class variable that could referenced by parent Distro.create_group which could be overridden on your netbsd Distro class so we don't have to maintain a copy of create_group method in your distro